src/test/java/com/thebois/models/world/structures/HouseTests.java

```diff
@@ -23,30 +34,206 @@ public void getPositionSetInConstructor() {
     }
 
     @Test
-    public void getTypeEqualsHouse() {
+    public void getTypeReturnsCorrectStructureType() {
         // Arrange
         final Position position = new Position(123, 456);
-
-        // Act
         final IStructure structure = StructureFactory.createStructure(
             StructureType.HOUSE,
             position);
+        // Act
+
+        StructureType returnedType = structure.getType();
 
         // Assert
-        assertThat(structure.getType()).isEqualTo(StructureType.HOUSE);
+        assertThat(returnedType).isEqualTo(StructureType.HOUSE);
     }
 
     @Test
-    public void deepCloneableIsEqualToOriginal() {
+    public void housesAreEqualWhenSameInstance() {
+        // Arrange
+        final Position positionA = new Position(123, 456);
+        final Position positionB = new Position(456, 123);
+
+        // Act
+        final IStructure structureA = StructureFactory.createStructure(
+            StructureType.HOUSE,
+            positionA);
+        final IStructure structureB = StructureFactory.createStructure(
+            StructureType.HOUSE,
+            positionB);
+
+        // Assert
+        assertThat(structureA).isNotEqualTo(structureB);
+    }
+
+    private static Stream<Arguments> tryDeliverItemReturnsExpectedValueSource() {
+        return Stream.of(
+            Arguments.of(10, 10, ItemFactory.fromType(ItemType.ROCK), false),
+            Arguments.of(10, 10,  ItemFactory.fromType(ItemType.LOG), false),
+            Arguments.of(0, 0,  ItemFactory.fromType(ItemType.ROCK), true),
+            Arguments.of(12, 12,  ItemFactory.fromType(ItemType.ROCK), false),
+            Arguments.of(10, 9,  ItemFactory.fromType(ItemType.ROCK), true));
+    }
+
+    @ParameterizedTest
+    @MethodSource("tryDeliverItemReturnsExpectedValueSource")
+    public void tryDeliverItemReturnsExpectedValue(
+        final int totalLogs,
+        final int totalRocks,
+        final IItem itemToDeliver,
+        final boolean expectedResult) {
```

Comment:
Something to think about when using parameterized tests is to control your variables. And by that, I mean that you should only have as many free variables as you need to. In this case, you could have the delivered item be fixed, and only let the required amount of items vary. Or the other way around.

Reply:
Thanks, I will look into it. But wouldn't that mean I would need many more tests that have houses with different amounts of items delivered to them? Right now I can tell the test what state the house should be in and what result I can expect from this state when an item is delivered.

Reply:
Yeah, this way is for sure more compact and requires less tests.

The problem is that you're basically testing a lot of things at once, in a single test. So if this test fails, it has multiple reasons for why it could have failed. If it were split up into more specific cases, then when one of them fails you'll immediately be able to see what kind of functionality is not working.

Reply:
Yeah, there are both upsides and downsides to splitting them up. If you don't want to spend time on refactoring the tests, that's fine since we're time-constrained this week.